chain/actor.go

```diff
@@ -0,0 +1,240 @@
+package chain
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/lotus/chain/state"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/ipfs/go-cid"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/sentinel-visor/lens"
+	"github.com/filecoin-project/sentinel-visor/model"
+	visormodel "github.com/filecoin-project/sentinel-visor/model/visor"
+	"github.com/filecoin-project/sentinel-visor/tasks/actorstate"
+)
+
+type ActorStateProcessor struct {
+	node          lens.API
+	opener        lens.APIOpener
+	closer        lens.APICloser
```

Comment:
it seems a bit scary that each processor holds the closer, since that's often going to be a shared handle that closes it fully for many lenses.

Reply:
The closer here is for the API instance that is returned when opener is called. It's not the same as the closer created when the opener is created. That is deferred in the top level command.